posthog/hogql_queries/query_runner.py

```diff
@@ -63,6 +64,7 @@ class QueryResponse(BaseModel, Generic[DataT]):
     hasMore: Optional[bool] = None
     limit: Optional[int] = None
     offset: Optional[int] = None
+    samplingRate: Optional[SamplingRate] = None
```

Comment:
We have a `samplingFactor` in most (if not all?) query objects (such as `TrendsQuery` and `StickinessQuery`). Would prefer it if we either:

1. Updated all the query runners to use this new sampling rate, or..
2. Move this inside of the relevant Web Analytics query objects.

Worried that in things like `TrendsQueryRunner`, it won't be clear whether we should be using `self.samplingRate` or `self.query.samlingFactor`

Reply:
The trends queries won't have access to `self._sample_rate`, the web analytics sampling stuff lives inside the root web analytics query runner.

I didn't want to combine how web and trends do sampling, because they are quite different.

Trends: sampling factor is included in the request, no need to include it in the response.

Web: sampling is enabled in the request but the factor is calculated on the back end, and is included in the response so it can be shown to the user.

Also, afaict this `QueryResponse` is not the "real" query response type, those are in the schema files, this functions as more like the intersection of all query response types, and is not used outside of this file / its test file. It's only there to provide some safety to the response caching (so imo it's ok to add this here)

Reply:
Ahh, didn't realise this was in the `QueryResponse` class, and not the `QueryRunner` class - GitHub failed me there!